### PR TITLE
cleanups and correcly assign fieldref

### PR DESCRIPTION
**Pull request by colinsurprenant**
- code cleanups
- correctly assign field ref

relates to elastic/logstash#4264 and elastic/logstash#4191

### CODE REVIEW

File: lib/logstash/inputs/imap.rb [hunk 1]
```diff
@@ -129,21 +121,24 @@ def parse_mail(mail)
         # Assume we already processed the 'date' above.
         next if name == "Date"
 
-        case event[name]
+        case (field = event[name])
```
Comment (jordansissel) [inline lib/logstash/inputs/imap.rb]: Oh, I like this style. Good call.

**Comment by jordansissel** [comment]
LGTM

**Comment by colinsurprenant** [comment]
ah thanks! but merged before version bump! will do in another PR.

**Comment by colinsurprenant** [comment]
pushed version 2.0.3 with PR #14

**Pull request closed without merging**